platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,178 @@
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = Collections.emptyMap();
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        Configuration[] configs = configurationAdmin.listConfigurations(null);
+        if (configs == null) {
+            return;
+        }
+        this.pidFileMap = Arrays.stream(configs)
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile,
+                        // Use the latest value for the same key
+                        (pid1, pid2) -> pid2));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        try {
+            switch (configurationEvent.getType()) {
+            case CM_UPDATED:
+                LOGGER.debug("Handling update");
+                Configuration configuration = getConfiguration(configurationEvent);
+                if (configuration == null) {
+                    break;
+                }
+                handleUpdate(new FelixConfig(configuration));
+                break;
+            case CM_DELETED:
+                LOGGER.debug("Handling delete");
+                handleDelete(configurationEvent.getPid());
+                break;
+            default:
+                LOGGER.debug("Unknown configuration event type, taking no action");
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error reading or modifying configurations on disk: {}", e);
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param config the {@link Configuration} wrapped in a {@link FelixConfig} that arrived as an event.
+     * @throws IOException if an issue occurs persisting to disk.
+     */
+    private void handleUpdate(FelixConfig config) throws IOException {
+        String pid = config.getPid();
+        File felixFileNew = config.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        boolean filePropChanged = config.filePropChanged(felixFileOriginal);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
+            LOGGER.debug("Was not tracked and will not track pid {}", pid);
+            return;
+        }
+
+        if (filePropChanged) {
```

Comment:
✏️ Can we just make the call to the `config.filePropChanged()` method inline here instead of storing it in a local variable?